components/developer/jenkins/files/jenkins-standalone.xml

```diff
@@ -95,15 +101,36 @@
           </method_context>
         </exec_method>
 
+<!-- At the bare minimum, we want to restart and redeploy the WAR file and
+     its Jetty container when the service is refreshed (e.g. package update).
+     So we just kill it (if running) and hope SMF restart would do the rest.
```

Comment:
Feel free to improve iteratively. This is a bare minimum which does work, hands-off. Shutdown is graceful (JVM has a SIGTERM and processes it to stop jobs, etc.) though fast (probably does not wait for jobs to finish running, if anything is building).

Jenkins2 has a mode with workers/nodes/agents/slaves/whatever the PC name is nowadays, where the master can be restarted while remote jobs are running - and it reattaches to manage them. So this is not very disruptive.

Also there is some balance to be had about updating packages and not blocking `pkg` for hours wating for the service to restart leisurely (can untie this somehow else though, if needed in practice). And system shutdowns of different urgency have their say too. Going down for maintenance window, well prepared, is one thing. Powering off safely because UPS is ONBATT is another.

All this can be handled by a method script, but really is out of scope for weekly bumps of the core and can be handled by a separate PR to improve this proposed method, hanging in discussion for a few more weeks :)